morf-core/src/main/java/org/alfasoftware/morf/sql/ParallelQueryHint.java

```diff
@@ -1,27 +1,65 @@
 package org.alfasoftware.morf.sql;
 
-import java.util.Objects;
+import static java.lang.String.format;
+
+import java.util.Optional;
+
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+
 /**
  * Represents the hint for a parallel query.
  *
  * @author Copyright (c) Alfa Financial Software 2018
  */
 public final class ParallelQueryHint implements Hint {
 
+  private final Optional<Integer> degreeOfParallelism;
+
+  public ParallelQueryHint(Optional<Integer> degreeOfParallelism) {

```

Comment:
This constructor can take primative int. The caller knows that a degree is definitely being set.